src/orchestratord/src/controller/materialize/resources.rs

```diff
@@ -299,60 +299,130 @@ fn create_network_policies(
-                        namespace_selector: Some(LabelSelector {
-                            match_labels: Some(btreemap! {
-                                "kubernetes.io/metadata.name".into() => orchestratord_namespace.into(),
+                }),
+            },
+            // Allow traffic from orchestratord to environmentd in order to hit
+            // the promotion endpoints during upgrades
+            NetworkPolicy {
+                metadata: mz.managed_resource_meta(mz.name_prefixed("allow-orchestratord")),
+                spec: Some(NetworkPolicySpec {
+                    ingress: Some(vec![NetworkPolicyIngressRule {
+                        from: Some(vec![NetworkPolicyPeer {
+                            namespace_selector: Some(LabelSelector {
+                                match_labels: Some(btreemap! {
+                                    "kubernetes.io/metadata.name".into() => orchestratord_namespace.into(),
+                                }),
+                                ..Default::default()
                             }),
-                            ..Default::default()
-                        }),
-                        pod_selector: Some(LabelSelector {
-                            match_labels: Some(btreemap! {
-                                "materialize.cloud/app".into() => "orchestratord".into(),
+                            pod_selector: Some(LabelSelector {
+                                match_labels: Some(btreemap! {
+                                    "materialize.cloud/app".into() => "orchestratord".into(),
+                                }),
+                                ..Default::default()
                             }),
                             ..Default::default()
-                        }),
-                        ..Default::default()
-                    }]),
-                    ports: Some(vec![NetworkPolicyPort {
-                        port: Some(IntOrString::Int(config.environmentd_internal_http_port)),
-                        protocol: Some("TCP".to_string()),
+                        }]),
+                        ports: Some(vec![NetworkPolicyPort {
+                            port: Some(IntOrString::Int(config.environmentd_internal_http_port)),
+                            protocol: Some("TCP".to_string()),
+                            ..Default::default()
+                        }]),
                         ..Default::default()
                     }]),
+                    policy_types: Some(vec!["Ingress".to_owned()]),
+                    ..Default::default()
+                }),
+            },
+        ]);
+    }
+    if config.network_policies.ingress_enabled {
+        network_policies.extend([NetworkPolicy {
+            metadata: mz.managed_resource_meta(mz.name_prefixed("sql-and-http-ingress")),
+            spec: Some(NetworkPolicySpec {
+                ingress: Some(vec![NetworkPolicyIngressRule {
```

Comment:
we probably want this policy to be restricted to environmentd (or eventually balancer) pods, i think?

Reply:
I think eventually we want to restrict it to the balancer, but that wasn't available yet.